src/components/services-view/services-view.controller.ts

```diff
@@ -1,156 +1,173 @@
-          return service.metadata.subCategory === subCategory;
+
+    public filterByCategory(category: string, subCategory: string, updateSubCategories: boolean) {
+        if (category === 'all' && subCategory === 'all') {
+            this.ctrl.filteredItems = this.ctrl.allItems;
         } else {
-          return  service.metadata.category === category && service.metadata.subCategory === subCategory;
+            this.ctrl.filteredItems = this.$filter('filter')(this.ctrl.allItems, (item: any) => {
+                if (category !== 'all' && subCategory === 'all') {
+                    return item.category === category;
+                } else if (category === 'all' && subCategory !== 'all') {
+                    return item.subCategory === subCategory;
+                } else {
+                    return  item.category === category && item.subCategory === subCategory;
+                }
+            });
         }
-      });
+
+        if (updateSubCategories) {
+            this.ctrl.subCategories = this.getSubCategories(category);
+        }
+
+        this.ctrl.expandSubCatRow = this.getRowOfSubCategory(subCategory);
+
+        this.ctrl.currentFilter = category;
+        this.ctrl.currentSubFilter = (subCategory !== undefined) ? subCategory : 'all';
     }
 
-    if (updateSubCategories) {
-      this.ctrl.subCategories = this.getSubCategories(category);
+    public toggleExpand(subCategory: string) {
+        if (this.ctrl.currentSubFilter === subCategory) {
+            this.ctrl.currentSubFilter = null;
+            this.ctrl.expandSubCatRow = ServicesViewController.NO_SUBCATEGORY_ROW;
+        } else {
+            this.ctrl.filterByCategory(this.ctrl.currentFilter, subCategory, false);
+        }
     }
 
-    this.ctrl.expandSubCatRow = this.getRowOfSubCategory(subCategory);
+    public getSubCategories(category: string) {
+        let subCats = [{id: 'all', label:  'All'}];
+        this.ctrl.categories.map(categoryObj => {
+            if (category === 'all' || category === categoryObj.id) {
+                subCats = subCats.concat(categoryObj.subCategories);
+            }
+        });
+
+        subCats = this.makeRows(subCats);
 
-    this.ctrl.currentFilter = category;
-    this.ctrl.currentSubFilter = (subCategory !== undefined) ? subCategory : 'all';
-  }
+        return subCats;
+    };
 
-  public toggleExpand(subCategory: string) {
-    if (this.ctrl.currentSubFilter === subCategory) {
-      this.ctrl.currentSubFilter = null;
-      this.ctrl.expandSubCatRow = ServicesViewController.NO_SUBCATEGORY_ROW;
-    } else {
-      this.ctrl.filterByCategory(this.ctrl.currentFilter, subCategory, false);
+    public handleClick(item: any, e: any) {
+        item.ctrl.serviceToOrder = item;
+        item.ctrl.openOrderingPanel();
+    };
+
+    public openOrderingPanel() {
+        this.ctrl.orderingPanelvisible = true;
+    };
+
+    public closeOrderingPanel = () => {
+        this.ctrl.orderingPanelvisible = false;
```

Comment:
Hmm I assumed the overlay was being done as a modal (like uibModal), I twill take a look at the overlay code and get back to you on this one.